pkg/cmd/pr/merge/merge.go

```diff
@@ -253,37 +269,50 @@ func (m *mergeContext) merge() error {
 		setCommitBody: m.opts.BodySet,
 	}
 
+	if m.shouldAddToMergeQueue() && !m.opts.InteractiveMode {
+		_ = m.warnf("%s A merge strategy cannot be set when adding a pull request to the merge queue\n", m.cs.FailureIconWithColor(m.cs.Red))
+		return cmdutil.SilentError
+	}
+
 	// get user input if not already given
 	if m.opts.InteractiveMode {
-		apiClient := api.NewClientFromHTTP(m.httpClient)
-		r, err := api.GitHubRepo(apiClient, m.baseRepo)
-		if err != nil {
-			return err
-		}
-
-		payload.method, err = mergeMethodSurvey(r)
-		if err != nil {
-			return err
-		}
-
-		m.deleteBranch, err = deleteBranchSurvey(m.opts, m.crossRepoPR, m.localBranchExists)
-		if err != nil {
-			return err
-		}
+		if m.shouldAddToMergeQueue() {
+			// auto merge will either enable auto merge or add to the merge queue
+			payload.auto = true
+		} else {
+			apiClient := api.NewClientFromHTTP(m.httpClient)
+			r, err := api.GitHubRepo(apiClient, m.baseRepo)
+			if err != nil {
+				return err
+			}
 
-		allowEditMsg := payload.method != PullRequestMergeMethodRebase
-		for {
-			action, err := confirmSurvey(allowEditMsg)
+			payload.method, err = mergeMethodSurvey(r)
 			if err != nil {
-				return fmt.Errorf("unable to confirm: %w", err)
+				return err
 			}
 
-			submit, err := confirmSubmission(m.httpClient, m.opts, action, &payload)
+			m.deleteBranch, err = deleteBranchSurvey(m.opts, m.crossRepoPR, m.localBranchExists)
 			if err != nil {
 				return err
 			}
-			if submit {
-				break
+
+			allowEditMsg := (payload.method != PullRequestMergeMethodRebase) &&
+				// merging as admin ignores the merge queue if it is required
+				(!m.mergeQueueRequired || m.opts.UseAdmin)
```

Comment:
This condition looks like an inverse of `m.shouldAddToMergeQueue()`. Since we are in the `else` branch of `if m.shouldAddToMergeQueue()`, wouldn't the inverse always be true? 

In the web UI, can the merge commit subject or message be additionally edited by the user around the time of adding to a merge queue?

Reply:
this was updated, but to answer the question there is no editing allowed with merge queue